src/uu/pgrep/src/pgrep.rs

```diff
@@ -0,0 +1,278 @@
+            Some(it)
+        } else {
+            None
+        }
+    };
+
+    let mut result = Vec::new();
+
+    for ele in walk_pid() {
+        if let Some(ele) = predicate_closure(ele) {
+            result.push(ele)
+        }
+    }
+
+    result
+}
+
+// Make -o and -n as a group of args
+fn collect_oldest_newest(matches: &ArgMatches, patterns: &[String]) -> Option<Vec<PidEntry>> {
+    let flag_newest = matches.get_flag("newest");
+    let flag_oldest = matches.get_flag("oldest");
+    let arg_older = matches.get_one::<u64>("older").unwrap();
+
+    if flag_newest != flag_oldest {
+        // Only accept one pattern.
+        if !patterns.is_empty() && patterns.len() != 1 {
+            println!("pgrep: only one pattern can be provided");
+            return None;
+        }
+
+        // Processing pattern
+        let result = collect_pid(matches, patterns);
+
+        let mut result = {
+            let mut vec = Vec::with_capacity(result.len());
+            for mut ele in result {
+                if ele.start_time().unwrap() >= *arg_older {
+                    vec.push(ele)
+                }
+            }
+            vec
+        };
+
+        result.sort_by(|a, b| {
+            if let (Ok(b), Ok(a)) = (
+                b.to_owned().borrow_mut().start_time(),
+                a.to_owned().borrow_mut().start_time(),
+            ) {
+                b.cmp(&a)
+            } else {
+                Ordering::Equal
+            }
+        });
+
+        // Check is empty
+        if result.is_empty() {
+            return Some(Vec::new());
+        }
+
+        let mut entry = if flag_newest {
+            result.first()
+        } else {
+            result.last()
+        }
+        .cloned()
+        .unwrap();
+
+        let sort = |start_time: u64| {
+            let mut result = result
+                .into_iter()
+                .filter(|it| (*it).to_owned().borrow_mut().start_time().is_ok())
+                .filter(move |it| (*it).to_owned().borrow_mut().start_time().unwrap() == start_time)
+                .collect::<Vec<_>>();
+
+            result.sort_by(|a, b| {
+                if let (Ok(b), Ok(a)) = (
+                    (*b).to_owned().borrow_mut().start_time(),
+                    (*a).to_owned().borrow_mut().start_time(),
+                ) {
+                    b.cmp(&a)
+                } else {
+                    Ordering::Equal
+                }
+            });
+
+            result
+        };
+
+        // TODO: make sure it can be multiple result
+        return Some(vec![sort(entry.start_time().unwrap())
+            .first()
+            .unwrap()
+            .clone()
+            .clone()]);
+    }
+
+    None
+}
+
+fn collect_normal_pid(matches: &ArgMatches, patterns: &[String]) -> Option<Vec<PidEntry>> {
+    let result = collect_pid(matches, patterns);
+
+    // Normal output
+    let result = result.into_iter().collect::<Vec<_>>();
+
+    Some(result)
+}
```

Comment:
I don't understand this function. What does "normal" mean in this context? Why is it needed? And why does it return an `Option`, even though it never returns `None`?